kerastuner/engine/trial.py

```diff
@@ -17,34 +17,39 @@
 from __future__ import division
 from __future__ import print_function
 
-import os
+import hashlib
 import json
+import os
+import random
+import time
 
+from kerastuner.engine import stateful
 from . import metrics_tracking
 from . import hyperparameters as hp_module
-from . import execution as execution_module
 from ..abstractions import display
 from ..abstractions.tensorflow import TENSORFLOW_UTILS as tf_utils
 
 
-class Trial(object):
+class TrialStatus:
+    RUNNING = "RUNNING"
+    IDLE = "IDLE"
```

Comment:
What does IDLE mean? When can a trial be in this state?

Reply:
This is used in HyperBand, when a Trial for the next bracket is requested before all the other Trials in the bracket have completed. It means the Tuner should wait and ask the Oracle again